language-detection-fasttext/src/main/java/com/hpe/caf/languagedetection/fasttext/FastTextWrapper.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2015-2019 Micro Focus or one of its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hpe.caf.languagedetection.fasttext;
+
+import com.hpe.caf.languagedetection.DetectedLanguage;
+import static com.hpe.caf.languagedetection.fasttext.FastTextDetector.MIN_RELIABILITY;
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import jep.JepException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class FastTextWrapper {
+    private static final Logger LOGGER = LoggerFactory.getLogger(FastTextWrapper.class);
+    private static final double MIN_LINE_ACCURACY = 0.95;
+    private static final int WORDS_PER_SENTENCE = 3;
+    private final String langPrefix = "__label__";
+    
+    public LDResult detect(final String document) {
+        final List<String> lines = new ArrayList<>();
+        for (final String line : document.split("\n")) {
```

Comment:
This code is pretty heavy on memory, by the time the string is split into lines we effectively have two copies of the content. The content could be 100mb.
Is there another way to provide the input to fastext?

Reply:
Updated, not making a copy of any other substring than one which is necessary for detection.